python/pip_install/private/generate_whl_library_build_bazel.bzl

```diff
@@ -126,6 +145,87 @@ def _render_list_and_select(deps, deps_by_platform, tmpl):
     else:
         return "{} + {}".format(deps, deps_by_platform)
 
+def _render_config_settings(dependencies_by_platform):
+    py_version_by_os_arch = {}
+    for p in dependencies_by_platform:
+        # p can be one of the following formats:
+        # * @platforms//os:{value}
+        # * @platforms//cpu:{value}
+        # * @//python/config_settings:is_python_3.{minor_version}
+        # * {os}_{cpu}
+        # * cp3{minor_version}_{os}_{cpu}
```

Comment:
FYI, I had to move the rendering of the config settings outside the main loop because it was getting complex. However, this PR can generate the right select for `pylint`:
```starlark
    deps = [
        "@pip_39_astroid//:pkg",
        "@pip_39_dill//:pkg",
        "@pip_39_isort//:pkg",
        "@pip_39_mccabe//:pkg",
        "@pip_39_platformdirs//:pkg",
        "@pip_39_tomlkit//:pkg",
    ] + select(
        {
            "@@rules_python~override//python/config_settings:is_python_3.10": ["@pip_39_tomli//:pkg"],
            "@@rules_python~override//python/config_settings:is_python_3.9": [
                "@pip_39_tomli//:pkg",
                "@pip_39_typing_extensions//:pkg",
            ],
            ":is_cp310_windows_anyarch": [
                "@pip_39_colorama//:pkg",
                "@pip_39_dill//:pkg",
                "@pip_39_tomli//:pkg",
            ],
            ":is_cp311_windows_anyarch": [
                "@pip_39_colorama//:pkg",
                "@pip_39_dill//:pkg",
            ],
            ":is_cp39_windows_anyarch": [
                "@pip_39_colorama//:pkg",
                "@pip_39_dill//:pkg",
                "@pip_39_tomli//:pkg",
                "@pip_39_typing_extensions//:pkg",
            ],
            "//conditions:default": [],
        },
    ),
```

Disregard the fact that the conditions and the target repos should match - this can be done as a followup item. I am not sure yet what design would be best. This versioned select is only enabled if the user specifies multiple target python versions.